kernel/aster-nix/src/fs/file_table.rs

```diff
@@ -89,8 +90,12 @@ impl FileTable {
         };
 
         let min_free_fd = get_min_free_fd();
-        let entry = FileTableEntry::new(file, flags);
+        let entry = FileTableEntry::new(file.clone(), flags);
         self.table.put_at(min_free_fd, entry);
+
+        if let Some(inode_file) = file.downcast_ref::<InodeHandle>() {
+            inode_file.release_range_locks();
```

Comment:
Could you show me the reference that says range locks should be released when a fd is duplicated?

Reply:
This is a mistake, I think. The duplicated file descriptors should inherit the locks. The following text elaborates that these lock operations are performed through the same file descriptor or its copies.

> Open file description locks placed via the same open file
>       description (i.e., via the same file descriptor, or via a
>       duplicate of the file descriptor created by [fork(2)](https://man7.org/linux/man-pages/man2/fork.2.html), [dup(2)](https://man7.org/linux/man-pages/man2/dup.2.html),
>       fcntl() F_DUPFD, and so on) are always compatible: if a new lock
>       is placed on an already locked region, then the existing lock is
>       converted to the new lock type.  (Such conversions may result in
>       splitting, shrinking, or coalescing with an existing lock as
>       discussed above.)